### PR TITLE
Fix calendar card height

### LINKED ISSUE #26990: Calendar card overflows set layout height
### Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

Image

### Describe the behavior you expected

Card should have min and max set based on view set (list should allow less than month view etc)

### Steps to reproduce the issue

1. Create calendar card
2. Set layout height

### What version of Home Assistant Core has the issue?

core-2025.9.1

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue?

Zen (FF) / Chrome

### Which operating system are you using to run this browser?

Arch Linux

### State of relevant entities

```txt

```

### Problem-relevant frontend configuration

```yaml

```

### JavaScript errors shown in your browser console/inspector

```txt

```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by timmo001**
## Proposed change

Sets boundaries for the calendar card as well as measuring the height for the calendar view.

image

image

image

Appears to scale well with the other views also:

image
image

Masonry and sidebar layouts (default 400px):

image

image

Panel layout (fills page):

image

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #26990
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/hui-calendar-card.ts [hunk 1]
```diff
@@ -187,6 +208,26 @@ export class HuiCalendarCard extends LitElement implements LovelaceCard {
...
+      0,
+      card.clientHeight - headerHeight - fcHeaderHeight - paddingBottom
+    );
+    const newHeight = `${Math.round(available)}px`;
+    if (newHeight !== this._calendarHeight) {
+      this._calendarHeight = newHeight;
+    }
   }
 
   private async _attachObserver(): Promise<void> {
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-calendar-card.ts:212]: (We should make this a `ResizeController` instead)

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-calendar-card.ts:212]: https://github.com/home-assistant/frontend/issues/27053

File: src/panels/lovelace/cards/hui-calendar-card.ts [hunk 2]
```diff
@@ -187,6 +208,26 @@ export class HuiCalendarCard extends LitElement implements LovelaceCard {
       return;
     }
     this._narrow = card.offsetWidth < 870;
+
+    const header = this.shadowRoot!.querySelector<HTMLDivElement>(".header");
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-calendar-card.ts]: If the card has a fixed height, can't we just set the height to 100% to fill it?

And does this also work for when it is not used in a section view, so when there isnt a fixed height?

Comment (timmo001) [reply src/panels/lovelace/cards/hui-calendar-card.ts]: Turns out the fix was just this, remove the variable and replace it with a height to fill the rest of the card with an allowance for the header padding

File: src/panels/lovelace/cards/hui-calendar-card.ts [hunk 3]
```diff
@@ -222,7 +235,7 @@ export class HuiCalendarCard extends LitElement implements LovelaceCard {
     }
 
     ha-full-calendar {
-      --calendar-height: 400px;
+      height: calc(100% - 16px);
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-calendar-card.ts:250]: Did you also test this in a none section view? Does this also work for the masonry view? I think we need to do this conditional like we do for some other cards?

Comment (timmo001) [reply src/panels/lovelace/cards/hui-calendar-card.ts:250]: Ah yes masonry has issues
image

File: src/panels/lovelace/cards/hui-calendar-card.ts [hunk 4]
```diff
@@ -222,7 +242,12 @@ export class HuiCalendarCard extends LitElement implements LovelaceCard {
     }
 
     ha-full-calendar {
-      --calendar-height: 400px;
+      height: 400px;
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-calendar-card.ts]: Why did we use a variable before and not anymore?

Comment (timmo001) [reply src/panels/lovelace/cards/hui-calendar-card.ts]: Not entirely sure, looks like it originally came from https://github.com/home-assistant/frontend/pull/18394/files

The only usage of `--calendar-height` is here: https://github.com/home-assistant/frontend/blob/9fc6386b7971491ff07a3d9024d35ac624651e61/src/panels/calendar/ha-full-calendar.ts#L507

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-calendar-card.ts]: So the difference is that we used to set the height if the calendar itself (just the table), and now we set the height of the component, which includes the controls on top. So we made the calendar smaller than it was before, and it now depends on the size of the header. Not sure which we prefer tbh, but maybe better not to change it if there is no need.

Comment (timmo001) [reply src/panels/lovelace/cards/hui-calendar-card.ts]: Restored original, height is still set correctly on cards, grid, masonry, calendar panel etc

**Review by bramkragten — APPROVED** [review]
